Framework/DataHandling/src/LoadILLPolarizedDiffraction.cpp

```diff
@@ -0,0 +1,411 @@
+                                                 FileProperty::Load, ".nxs"),
+                  "File path of the data file to load");
+  declareProperty(std::make_unique<WorkspaceProperty<API::WorkspaceGroup>>(
+                      "OutputWorkspace", "", Direction::Output),
+                  "The output workspace.");
+  std::vector<std::string> positionCalibrationOptions{"None", "Nexus",
+                                                      "YIGFile"};
+  declareProperty(
+      "PositionCalibration", "None",
+      std::make_shared<StringListValidator>(positionCalibrationOptions),
+      "Select the type of pixel position calibration. If None, the pixel "
+      "positions are read from IDF file. If Nexus, the positions are read from "
+      "Nexus file. If YIGFile, then the calibration twotheta data is loaded "
+      "from a user-defined calibration file.");
+
+  declareProperty(std::make_unique<FileProperty>(
+                      "YIGFilename", "", FileProperty::OptionalLoad, ".xml"),
+                  "File path of the YIG calibration data file to load");
+  setPropertySettings("YIGFilename",
+                      std::make_unique<Kernel::EnabledWhenProperty>(
+                          "PositionCalibration", IS_EQUAL_TO, "YIGFile"));
+}
+
+/**
+ * Executes the algorithm.
+ */
+void LoadILLPolarizedDiffraction::exec() {
+
+  Progress progress(this, 0, 1, 2);
+
+  m_filename = getPropertyValue("Filename");
+
+  m_outputWorkspace = std::make_shared<API::WorkspaceGroup>();
+  setProperty("OutputWorkspace", m_outputWorkspace);
+
+  progress.report("Loading the detector polarization analysis data");
+  loadData();
+
+  progress.report("Loading the metadata");
+  loadMetaData();
+}
+
+/**
+ * Loads the polarized detector data, sets up workspaces and labels
+ *  according to the measurement type and data dimensions
+ */
+void LoadILLPolarizedDiffraction::loadData() {
+
+  // open the root entry
+  NXRoot dataRoot(m_filename);
+
+  // read each entry
+  for (auto entryNumber = 0;
+       entryNumber < static_cast<int>((dataRoot.groups().size()));
+       entryNumber++) {
+    NXEntry entry = dataRoot.openEntry("entry" + std::to_string(entryNumber));
+    m_instName = entry.getString("D7/name");
+
+    std::string start_time = entry.getString("start_time");
+    start_time = m_loadHelper.dateTimeInIsoFormat(start_time);
+
+    // check the mode of measurement and prepare axes for data
+    std::vector<double> axis;
+    NXInt acquisitionMode = entry.openNXInt("acquisition_mode");
+    acquisitionMode.load();
+    if (acquisitionMode[0] == 1) {
+      NXFloat timeOfFlightInfo =
+          entry.openNXFloat("D7/Detector/time_of_flight");
+      timeOfFlightInfo.load();
+      auto channelWidth = static_cast<double>(timeOfFlightInfo[0]);
+      m_numberOfChannels = size_t(timeOfFlightInfo[1]);
+      auto tofDelay = timeOfFlightInfo[2];
+      for (auto ii = 0; ii <= static_cast<int>(m_numberOfChannels); ii++) {
```

Comment:
We try to avoid variable names that do not mean anything, perhaps ii would be better as channel_no or channel_index?